### PR TITLE
Port 0x Api

### PR DESCRIPTION
**Pull request by nlordell**
Similar to #1234, this PR ports the 0x API to the `solvers` crate.

Nothing too special, although we make use of the `reqwest::RequestBuilder::query(q: impl serde::Serialize)` to avoid manual serialization code 🎉.

Additionally, I added some (hopefully) more complete documentation to the DTOs with links to the online documentation. I also removed fields that we don't need/use to keep the DTOs as slim as possible.

### Test Plan

No E2E test quite yet. However, a manual test can be run:

<details><summary><code>manual-test.patch</code></summary>

```diff
diff --git a/crates/solvers/src/infra/dex/zeroex/mod.rs b/crates/solvers/src/infra/dex/zeroex/mod.rs
index 287385da..d2ff7ce3 100644
--- a/crates/solvers/src/infra/dex/zeroex/mod.rs
+++ b/crates/solvers/src/infra/dex/zeroex/mod.rs
@@ -190,3 +190,55 @@ impl From<dto::Error> for Error {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[tokio::test]
+    async fn manual() {
+        shared::tracing::initialize_reentrant("solvers=trace");
+
+        let dex = ZeroEx {
+            client: reqwest::Client::new(),
+            endpoint: DEFAULT_URL.parse().unwrap(),
+            settlement: eth::ContractAddress(
+                "0x9008D19f58AAbD9eD0D60971565AA8510560ab41"
+                    .parse()
+                    .unwrap(),
+            ),
+            defaults: dto::Query {
+                excluded_sources: vec![
+                    "Uniswap".to_owned(),
+                    "SushiSwap".to_owned(),
+                    "Curve".to_string(),
+                ],
+                ..Default::default()
+            },
+        };
+
+        let swap = dex
+            .swap(
+                &dex::Order {
+                    sell: eth::TokenAddress(
+                        "0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2"
+                            .parse()
+                            .unwrap(),
+                    ),
+                    buy: eth::TokenAddress(
+                        "0xE41d2489571d322189246DaFA5ebDe1F4699F498"
+                            .parse()
+                            .unwrap(),
+                    ),
+                    side: order::Side::Sell,
+                    amount: dex::Amount::new(1_000_000_000_000_000_000_u128.into()),
+                },
+                &dex::Slippage::one_percent(),
+                auction::GasPrice(eth::Ether(15_000_000_000_u128.into())),
+            )
+            .await
+            .unwrap();
+
+        panic!("{swap:#?}");
+    }
+}

```

</details>

### CODE REVIEW

File: crates/solvers/src/infra/dex/zeroex/dto.rs [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+/// A Ox API quote response.
+#[serde_as]
+#[derive(Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Quote {
+    /// The address of the contract to call in order to execute the swap.
+    pub to: H160,
+
+    #[serde_as(as = "serialize::Hex")]
+    /// The swap calldata.
```
Comment (MartinquaXD) [inline crates/solvers/src/infra/dex/zeroex/dto.rs:99]: To make it consistent with the other fields.
```suggestion
    /// The swap calldata.
    #[serde_as(as = "serialize::Hex")]
```

File: crates/solvers/src/infra/dex/zeroex/dto.rs [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    pub data: Vec<u8>,
+
+    /// The Ether value to use in order to execute the swap.
+    #[serde_as(as = "serialize::U256")]
+    pub value: U256,
+
+    // The amount of sell token (in atoms) that would be sold in this swap.
```
Comment (MartinquaXD) [inline crates/solvers/src/infra/dex/zeroex/dto.rs:106]:
```suggestion
    /// The amount of sell token (in atoms) that would be sold in this swap.
```

File: crates/solvers/src/infra/dex/zeroex/dto.rs [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+    #[serde_as(as = "serialize::U256")]
+    pub sell_amount: U256,
+
+    // The amount of buy token (in atoms) that would be bought in this swap.
```
Comment (MartinquaXD) [inline crates/solvers/src/infra/dex/zeroex/dto.rs:110]:
```suggestion
    /// The amount of buy token (in atoms) that would be bought in this swap.
```

**Review by MartinquaXD — APPROVED** [review]

**Review by vkgnosis — APPROVED** [review]
struct for query parameters is nice

**Pull request merged**